Add contributing guide

s-weigand opened the pull request:
This is my current draft, on what could be the contributing guide.

s-weigand committed:
added contribution guide

s-weigand committed:
formatted lines to be shorter and added paragraph about testing

s-weigand committed:
changed folder structure so, docs will build

see #32 and #9

s-weigand committed:
added symlink to CONTRIBUTING.md in the docs

s-weigand committed:
changed folder structure in docs to be more consistent with the pojects

s-weigand committed:
added docs/contributing.md symlink file  to .gitignore

s-weigand committed:
added examples and contributing to the docs toctree

s-weigand committed:
added .gitattributes so symlinks will work properly

s-weigand committed:
added cheat sheets and tutorials

s-weigand committed:
added more content to the contributing guide

s-weigand committed:
ignore link check to dynamically rendered anchors in gists

s-weigand committed:
renamed 'refresh_symlinks_windows.sh' to 'refresh_symlinks.sh'

and made link generation less aggressive

s-weigand committed:
added guide to temporarily add conda to path

s-weigand committed:
restructured main TOC

s-weigand committed:
Made guide to create the docs more detailed

s-weigand committed:
fixed markdown style violation recommended by codacy

s-weigand committed:
removed unneeded quotes for 'use_conda.bat', which actually broke the 

functionality

s-weigand committed:
refined description on how PATHS and conda play together

s-weigand committed:
Changed guide to install make into gitbash.

s-weigand committed:
Improved details about, how to contribute

s-weigand committed:
added missing link urls and fixed some typos

s-weigand committed:
refined contributing guide

s-weigand committed:
refined 'Using Anaconda only if needed' guide

s-weigand committed:
removed unneeded get info in url to tutorial link

s-weigand committed:
Made heading style consistent

s-weigand committed:
changed 'Note' parts to proper Note boxes in sphinx

redbluee commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,278 @@
+<p>
+For the docs to be build it is mandatory that you use a conda installation of python or at least have conda installed.
+This is due to the fact that the notebook inclusion in the docs utilizes the tool [pandoc](https://pandoc.org/).
+Even so `pandoc` is present in many package manager repository indices, this is mostly an too old version,
```
typo: "... this is mostly a too old version"

s-weigand replied:
fixed

s-weigand committed:
applied requested change

s-weigand commented:
Just saw I forgot to add the styleguide (#25). 😅

jsnel commented on .gitattributes:
```diff
@@ -0,0 +1,7 @@
+*	text=lf
+*.bat	text eol=crlf
+
+docs/readme.md	symlink=file
+docs/contributing.md	symlink=file
+docs/material/cheat_sheets.md	symlink=file
+docs/material/tutorials.md	symlink=file
```
It seem docs/readme.md, docs/contributing.md are symlinks
but docs/material/cheat_sheets.md and docs/material/tutorials.md seem like actual files.

s-weigand replied:
Just freshly cloned the repo on a WSL shell and all files are symlinks as they should be.

jsnel commented on .gitignore:
```diff
@@ -111,6 +111,9 @@ docs/**/**/images/
 
 # this is just a symlinks and needs no changing
 docs/readme.md
+docs/contributing.md
+docs/material/cheat_sheets.md
+docs/material/tutorials.md
```
Should the files in docs/material be in .gitignore?

s-weigand replied:
I had some problems with symlinks on Windows, especially in combination with the old version of GitKraken, which replaced symlinks with normal files containing the relative path when I pulled changes.
This is also why `docs/refresh_symlinks.sh` exists, but after fixing the symlinks GitKraken marked them as changed.
The command I was looking for is `git update-index --no-skip-worktree <file_path>`.
Updated `docs/refresh_symlinks.sh` to include it and removed the files from `.gitignore`.

jsnel commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,278 @@
+
+Look through the [GitHub issues](https://github.com/students-teach-students/python-tools-for-students/issues) for bugs. Anything tagged with "bug"
+and "help wanted" is open to whoever wants to implement it.
+You might also encounter typos, spelling and grammar errors, we appreciate all help,
```
easy on the punctuation: https://www.thepunctuationguide.com/comma.html

s-weigand replied:
Hope it's better now 😅

jsnel commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,278 @@
+### Write `TL;DR`'s or Examples
+
+We will leave the writing of `TL;DR`'s and examples mostly to the community,
+since this is the perfect opportunity, for **you** to get involved.
```
less commas

s-weigand replied:
Hope it's better now 😅

jsnel commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,278 @@
+</p>
+</div>
+
+<div class="admonition note">
```
strange class name
admonition = a firm warning or reprimand.

s-weigand replied:
That classname comes from `sphinx` and is how the `.. note::` derective in `*.rst` is rendered in `html`.

jsnel commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,278 @@
+<p class="admonition-title">Note for Windows users:</p>
+<p>
+
+If you want the care free package of a 'properly' configured Posix like shell,
+just install [`cmder full`](https://cmder.net/) with [get-cmder](https://github.com/s-weigand/get-cmder).
```
Windows users may not know (or care) what a posix shell is, so how would they know if this is relevant to them?

s-weigand replied:
True that.
Maybe change it to:
... Posix like shell (more powerful and feature rich command line) ...

s-weigand replied:
@jsnel what do you think about this change?

jsnel commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,278 @@
+</p>
+
+```bash
+$ conda install pandoc
````
did you intend to leave out -c conda-forge here?

s-weigand replied:
`pandoc` also is on the official anaconda channel.
https://anaconda.org/anaconda/pandoc
And the conda main channel
https://anaconda.org/main/pandoc

But since I link to the package on `conda-forge`, it would be more consistent if add it, also the versions there are updated more frequently.

jsnel commented on docs/refresh_symlinks.sh:
```diff
@@ -4,3 +4,6 @@
 export MSYS=winsymlinks:nativestrict
 ln -srf ../README.md readme.md
+ln -srf ../CONTRIBUTING.md contributing.md
+ln -srf ../material/cheat_sheets.md material/cheat_sheets.md
+ln -srf ../material/tutorials.md material/tutorials.md
```
seems like links to the files themselves

s-weigand replied:
They link from `<repo_root>/material` to `<repo_root>/docs/material`

jsnel commented on material/tutorials.md:
```diff
@@ -0,0 +1,11 @@
+# Tutorials
+
+## Python
```
Perhaps also consider something like https://realpython.com/ (they have a ton of free resources, but (for the paid user) also really great (video) courses.

s-weigand replied:
Added it and I also love their content 😄

jsnel reviewed (commented):
Reviewed ok, minor comments but none that should delay the merge.

s-weigand committed:
updated 'refresh_symlinks.sh' to use 'git update-index --no-skip-worktree'

on symlink files, thus there is no more need have them in .gitignore, which didn't always work as intended anyway

s-weigand committed:
updated the pandoc installation command to use the conda-forge channel

s-weigand committed:
removed instructions for 'docs/authors.rst' since it was dropped

we use 'all-contributors' for that

s-weigand committed:
Removed the overused commas

s-weigand committed:
Removed broken link to itworldnew and replaced it with links to 

sololearn and programminghub

s-weigand committed:
Added tutorial website for python (realpython+sentdex)

s-weigand commented:
@all-contributors please add @jsnel for review

s-weigand committed:
added style guide

jsnel reviewed (approved):
Looks good.

s-weigand committed:
added explanation what a posix like shell is
